tests/test_check.py

```diff
@@ -423,3 +423,26 @@ def test_precondition_inline_comment_no_error(caplog) -> None:
     )
 
     assert "WARNING" not in [record.levelname for record in caplog.records]
+
+
+def test_check_all_truncates_messages_when_max_exceeded(capsys):
+    """
+    Test that check_all outputs only the max number of messages set when the max is exceeded.
+    """
+    import python_ta
+
+    python_ta.check_all(
+        # tests/fixtures/unused_imports.py contains 3 unused import errors
+        "tests/fixtures/unused_imports.py",
+        config={
+            "output-format": "pyta-plain",
+            "pyta-error-permission": "no",
+            "pyta-file-permission": "no",
+            "pyta-number-of-messages": 2,  # Max messages is set to two, so the last error of the same type is not outputed
+        },
+    )
+
+    output = capsys.readouterr().out
+
+    # Check that only two messages are outputted
+    assert "First 2 shown" in output
```

Comment:
This is good, but let's add another assertion to ensure that only two error occurrences are actually displayed. It may be easier to work with the JSON report than the plaintext report to accomplish this.